src/t_hash.c

```diff
@@ -2997,11 +2995,6 @@ static void hexpireGenericCommand(client *c, const char *cmd, long long basetime
             return;
         }
         expire *= 1000;
-    } else {
-        if (expire > (long long) EB_EXPIRE_TIME_MAX) {
-            addReplyErrorExpireTime(c);
-            return;
-        }
```

Comment:
@moticless  this judgment actually overlaps with the following `if (expire > (long long) EB_EXPIRE_TIME_MAX - basetime)` block.

Reply:
We can change this logic. and maybe add short comment.

Reply:
doen with [`d54bb77` (#13310)](https://github.com/redis/redis/pull/13310/commits/d54bb77351e794189917b7f87e695ddaf59014c1)
not sure it's enough.